cmd/restore.go

```diff
@@ -446,3 +302,59 @@ func initRestoreClient(ctx context.Context, client *restore.Client, flagSet *fla
 
 	return nil
 }
+
+// RestorePrepareWork execute some prepare work before restore
+func RestorePrepareWork(ctx context.Context, client *restore.Client, mgr *conn.Mgr) ([]string, error) {
+	if client.IsOnline() {
+		return nil, nil
+	}
+	err := client.SwitchToImportMode(ctx)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	existSchedulers, err := mgr.ListSchedulers(ctx)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	needRemoveSchedulers := make([]string, 0, len(existSchedulers))
+	for _, s := range existSchedulers {
+		if _, ok := schedulers[s]; ok {
+			needRemoveSchedulers = append(needRemoveSchedulers, s)
+		}
+	}
+	return removePDLeaderScheduler(ctx, mgr, needRemoveSchedulers)
+}
+
+func removePDLeaderScheduler(ctx context.Context, mgr *conn.Mgr, existSchedulers []string) ([]string, error) {
+	removedSchedulers := make([]string, 0, len(existSchedulers))
+	for _, scheduler := range existSchedulers {
+		err := mgr.RemoveScheduler(ctx, scheduler)
+		if err != nil {
+			return nil, err
+		}
+		removedSchedulers = append(removedSchedulers, scheduler)
+	}
+	return removedSchedulers, nil
+}
+
+// RestorePostWork execute some post work after restore
+func RestorePostWork(ctx context.Context, client *restore.Client, mgr *conn.Mgr, removedSchedulers []string) error {
+	if client.IsOnline() {
+		return nil
+	}
+	err := client.SwitchToNormalMode(ctx)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	return addPDLeaderScheduler(ctx, mgr, removedSchedulers)
+}
+
+func addPDLeaderScheduler(ctx context.Context, mgr *conn.Mgr, removedSchedulers []string) error {
+	for _, scheduler := range removedSchedulers {
```

Comment:
Question. Since map is randomly ordered, the schedulers are added in random order as well. Is this OK for PD?

Reply:
AFAIK, yes, there is no dependency problem between each scheduler.